### DISCUSSION TITLE
What is expectedEntries for?

### DISCUSSION THREAD

**Discussion by FGasper** [maxatome/go-testdeep#245] [Q&A]
SuperMapOf’s documentation states that:
> expectedEntries can be nil, if no zero entries are expected and no TestDeep operators are involved.

However, I’m finding that both TestDeep operators and zero values work fine when included in the model.

So, what is a case where I need to give a non-nil expectedEntries?

Thank you for this library! Just the other day a couple coworkers of mine—who, like me, come from Perl—were talking about how we miss deep-comparison tools like Test::Deep in Go.

**Comment by maxatome** [comment]
Hi @FGasper,

Sorry for the delay, I didn't see your message in time.

You need `expectedEntries` when you want to use a TestDeep operator and your map values type is not `any` but `int` for example: https://go.dev/play/p/ALiLaAu2PAl

I am glad you and your coworkers finally found this Test::Deep "equivalent" and I hope you will promote it to gain some stars as it lacks of awareness... ;)

Best regards,

Max.
